examples/example_piecewise_target_function.py

```diff
@@ -0,0 +1,162 @@
+
+import matplotlib.pyplot as plt
+import numpy as np
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+
+# %%
+# We define a piecewise target function.  The function applies different
+# transformations to the input depending whether the input is less or greater
+# than zero. Thus to achieve high fitness,
+# an individual must make use of the if/else operator.
+def f_target(x):
+    return np.select([x >= 0, x < 0], [x ** 2 + 1.0, -x])
+
+
+# %%
+# Then we define the objective function for the evolution. It uses the
+# mean-squared error between the output of the expression represented by a given
+# individual and the target function evaluated on a set of pseudo-random points.
+def objective(individual, target_function, rng):
+    """Objective function of the regression task.
+
+    Parameters
+    ----------
+    individual : Individual
+        Individual of the Cartesian Genetic Programming Framework.
+    target_function : Callable
+        Target function.
+    rng: numpy.random.RandomState
+
+    Returns
+    -------
+    Individual
+        Modified individual with updated fitness value.
+    """
+    if individual.fitness is not None:
+        return individual
+
+    n_function_evaluations = 1000
+
+    f = individual.to_func()
+    y = np.empty(n_function_evaluations)
+    x = rng.uniform(-5, 5, size=n_function_evaluations)
+    for i, x_i in enumerate(x):
+        y[i] = f([x_i])[0]
+
+    loss = np.mean((target_function(x) - y) ** 2)
+    individual.fitness = -loss
+
+    return individual
+
+
+# %%
+# Next, we set up the evolutionary search. First, we define the parameters for
+# the population, the genomes of individuals, and the evolutionary
+# algorithm.
+population_params = {
+    "n_parents": 1,
+    "mutation_rate": 0.05,
+    "seed": 8188211,
+}
+
+genome_params = {
+    "n_inputs": 1,
+    "n_outputs": 1,
+    "n_columns": 20,
+    "n_rows": 1,
+    "levels_back": None,
+    "primitives": (cgp.IfElse, cgp.Mul, cgp.Add, cgp.Sub, cgp.ConstantFloat,),
+}
+
+ea_params = {"n_offsprings": 4, "tournament_size": 2, "n_processes": 2}
+
+evolve_params = {"max_generations": int(args["--max-generations"]), "min_fitness": 0.0}
+
+# create population that will be evolved
+pop = cgp.Population(**population_params, genome_params=genome_params)
+
+# create instance of evolutionary algorithm
+ea = cgp.ea.MuPlusLambda(**ea_params)
+
+# define callback for recording of fitness over generations
+history = {}
+history["fitness_champion"] = []
+history["champion_sympy"] = []
+
+
+def recording_callback(pop):
+    history["fitness_champion"].append(pop.champion.fitness)
+    try:
+        sympy_expression = pop.champion.to_sympy()
+    except TypeError:
+        sympy_expression = pop.champion.to_sympy(simplify=False)
+
+    history["champion_sympy"].append(sympy_expression)
+
+
+# the objective passed to evolve should only accept one argument,
+# the individual
+seed = 1000
+rng = np.random.RandomState(seed=seed)
```

Comment:
```suggestion
rng = np.random.RandomState(seed=population_params["seed"])
```
just to avoid introducing multiple seeds